docs: Document gaps to the community `cert-manager` (#379)
gardener/cert-management: README.md

## Demo quick start

1. Run dns-controller-manager with:

    ```bash
    ./dns-controller-manager --controllers=azure-dns --identifier=myOwnerId --disable-namespace-restriction
    ```

2. Ensure provider and its secret, e.g.

    ```bash
    kubectl apply -f azure-secret.yaml
    kubectl apply -f azure-provider.yaml
    ```

    - check with

         ```bash
         ▶ kubectl get dnspr
         NAME               TYPE        STATUS   AGE
         azure-playground   azure-dns   Ready    28m
         ```

3. Create test namespace

    ```bash
    kubectl create ns test
    ```

4. Run cert-controller-manager

    ```bash
    ./cert-controller-manager
    ```

5. Register user `some.user@mydomain.com` at Let's Encrypt

    ```bash
    kubectl apply -f examples/20-issuer-staging.yaml
    ```

    - check with

         ```bash
         ▶ kubectl get issuer
         NAME             SERVER                                                   EMAIL                    STATUS   TYPE   AGE
         issuer-staging   https://acme-staging-v02.api.letsencrypt.org/directory   some.user@mydomain.com   Ready    acme   8s
         ```

> 6. Request a certificate for `cert1.martin.test6227.ml`

Reviewer: ```suggestion
6. Request a certificate for `cert1.my-domain.com`
```

   Gardener's `cert-management` allows to configure the `followCNAME` option on the `Certificate` resource.

3. ⚠️ `CertificateRevocation`s are not supported in the `cert-manager` project.

4. ⚠️ Only Kubernetes `Gateway` resources can be annotated in the `cert-manager` project ([ref](https://cert-manager.io/docs/usage/gateway/)).
   Gardener's `cert-management` supports both Kubernetes `Gateway` and Istio `Gateway` resources.

5. ⚠️ It is not possible to annotate `Service` resources of type `LoadBalancer` in the `cert-manager` project ([ref](https://cert-manager.io/docs/usage/)).

6. ⛔️ There is no reuse of existing certificates in the `cert-manager` project.
   This means that if a certificate is requested multiple times, it will be issued multiple times.
   Gardener's `cert-management` reuses existing certificates if the common name and DNS names match.

7. ⛔️ The ACME DNS-01 challenge is represented through the custom resources `DNSEntry` and `DNSRecord` in the `cert-manager` project.

>    Gardener's `cert-management` uses the companion `dns-controller-manager` from [external-dns-management](https://github.com/gardener/external-dns-management) to solve DNS-01 challenges (in a possible separate DNS cluster). 

Reviewer 2: I feel like this sounds a bit more natural, maybe it misses the point though
```suggestion
   Gardener's `cert-management` uses the companion `dns-controller-manager` from [external-dns-management](https://github.com/gardener/external-dns-management) to solve DNS-01 challenges (possibly in a separate DNS cluster). 
```

   This would require developing a webhook `Issuer` for `cert-manager` to integrate with the `dns-controller-manager` ([ref](https://cert-manager.io/docs/configuration/acme/dns01/webhook/)).

> 8. ⛔️ Private keys for the ACME `Issuer` have to be stored in the same cluster as the `cert-manager` controller is running in ([ref](https://github.com/cert-manager/cert-manager/issues/756)).

Reviewer 3: ```suggestion
8. ⛔️ Private keys for the ACME `Issuer` have to be stored in the same cluster as the `cert-manager` controller is running in ([ref](https://github.com/cert-manager/cert-manager/issues/756)). This is an essential requirement for a hosted control-plane solution like Gardener, where issuers and especially their secrets should remain opaque to end-users.
```

Nit
